test/swar/BasicOperations.cpp

```diff
@@ -6,15 +6,146 @@
+static_assert(0x0808'0808'0808'0808ull == u64(broadcast<8>(SWAR<8, u64>(0x0000'0000'0000'0008ull))));
+
+static_assert(2 == lsb_index(1<<1));
+static_assert(4 == lsb_index(1<<3));
+static_assert(6 == lsb_index(1<<5));
+static_assert(9 == lsb_index(1<<8));
+static_assert(18 == lsb_index(1<<17));
+static_assert(31 == lsb_index(1<<30));
+
+static_assert(0x80880008 == greaterEqual<3>(SWAR<4, uint32_t>(0x3245'1027)).value());
+static_assert(0x88888888 == greaterEqual<0>(SWAR<4, uint32_t>(0x0123'4567)).value());
+static_assert(0x88888888 == greaterEqual<0>(SWAR<4, uint32_t>(0x7654'3210)).value());
+static_assert(0x00000008 == greaterEqual<7>(SWAR<4, uint32_t>(0x0123'4567)).value());
+static_assert(0x80000000 == greaterEqual<7>(SWAR<4, uint32_t>(0x7654'3210)).value());
+
+static_assert(0x8000'0080 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x1000'0010), SWAR<4, u32>(0x0111'1101)).value());
+static_assert(0x8000'0080 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x4333'3343), SWAR<4, u32>(0x4444'4444)).value());
+
+static_assert(0x8888'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x0000'0000)).value());
+static_assert(0x0888'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x1111'1111)).value());
+static_assert(0x0088'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x2222'2222)).value());
+static_assert(0x0008'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x3333'3333)).value());
+static_assert(0x0000'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x4444'4444)).value());
+static_assert(0x0000'0888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x5555'5555)).value());
+static_assert(0x0000'0088 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x6666'6666)).value());
+static_assert(0x0000'0008 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0123'4567), SWAR<4, u32>(0x7777'7777)).value());
+
+static_assert(0x8000'0000 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x0000'0000), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8800'0000 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x1111'1111), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8880'0000 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x2222'2222), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8888'0000 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x3333'3333), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8888'8000 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x4444'4444), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8888'8800 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x5555'5555), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8888'8880 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x6666'6666), SWAR<4, u32>(0x0123'4567)).value());
+static_assert(0x8888'8888 == greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(0x7777'7777), SWAR<4, u32>(0x0123'4567)).value());
```

Comment:
These are not clear tests, becau7se it is super hard to check humanly.
Perhaps:
```c++
s_a(greaterEqual....
0x4444'4444,
0x0123'4567) ==
0x8888'8000
``` 
^^^ the user has a spatial clue to compare